lib/go/test/flow_idtable_staking_test.go

```diff
@@ -2303,3 +2303,246 @@ func TestIDTableStaking(t *testing.T) {
+
+	// Fund each node and register each node for staking
+	for i := 0; i < numNodes; i++ {
+
+		// Fund the node account
+		mintTokensForAccount(t, b, nodeAddresses[i], "1000000.0")
+
+		// Register the node
+		committed[adminID] = registerNode(t, b, env,
+			nodeAddresses[i],
+			nodeSigners[i],
+			ids[i],
+			fmt.Sprintf("%0128s", ids[i]),
+			nodeNetworkingKeys[i],
+			nodeStakingKeys[i],
+			amountToCommit,
+			committed[ids[i]],
+			1,
+			false)
+
+	}
+
+	// Fund each delegator and register each one
+	// node 0 and 1 each get 5 delegators, but the other nodes get none
+	for i := 0; i < numDelegators; i++ {
+
+		// Fund the delegator account
+		mintTokensForAccount(t, b, delegatorAddresses[i], "1000000.0")
+
+		// Register the delegator
+		tx := createTxWithTemplateAndAuthorizer(b, templates.GenerateRegisterDelegatorScript(env), delegatorAddresses[i])
+		err := tx.AddArgument(cadence.String(ids[i/5]))
+		require.NoError(t, err)
+		signAndSubmit(
+			t, b, tx,
+			[]flow.Address{b.ServiceKey().Address, delegatorAddresses[i]},
+			[]crypto.Signer{b.ServiceKey().Signer(), delegatorSigners[i]},
+			false,
+		)
+
+		// Stake new tokens for the delegator
+		tx = createTxWithTemplateAndAuthorizer(b, templates.GenerateDelegatorStakeNewScript(env), delegatorAddresses[i])
+		_ = tx.AddArgument(CadenceUFix64("10000.0"))
+		signAndSubmit(
+			t, b, tx,
+			[]flow.Address{b.ServiceKey().Address, delegatorAddresses[i]},
+			[]crypto.Signer{b.ServiceKey().Signer(), delegatorSigners[i]},
+			false,
+		)
+	}
+
+	// Create an array of all the node IDs so they can be approved for staking
+	cadenceIDs := make([]cadence.Value, numNodes)
+	for i := 0; i < numNodes; i++ {
+		cadenceIDs[i] = CadenceString(ids[i])
+	}
+
+	// End the epoch, which marks everyone as staking
+	tx := createTxWithTemplateAndAuthorizer(b, templates.GenerateEndEpochScript(env), idTableAddress)
+	err := tx.AddArgument(cadence.NewArray(cadenceIDs))
+	require.NoError(t, err)
+	signAndSubmit(
+		t, b, tx,
+		[]flow.Address{b.ServiceKey().Address, idTableAddress},
+		[]crypto.Signer{b.ServiceKey().Signer(), IDTableSigner},
+		false,
+	)
+
+	t.Run("Should be able to set multiple nodes as inactive so their rewards will be withheld", func(t *testing.T) {
+
+		tx := createTxWithTemplateAndAuthorizer(b, templates.GenerateSetNonOperationalScript(env), idTableAddress)
+		err := tx.AddArgument(cadence.NewArray([]cadence.Value{CadenceString(ids[0]), CadenceString(ids[2])}))
+		require.NoError(t, err)
+		signAndSubmit(
+			t, b, tx,
+			[]flow.Address{b.ServiceKey().Address, idTableAddress},
+			[]crypto.Signer{b.ServiceKey().Signer(), IDTableSigner},
+			false,
+		)
+
+		result := executeScriptAndCheck(t, b, templates.GenerateGetNonOperationalListScript(env), nil)
+
+		idArray := result.(cadence.Array).Values
+		assert.Equal(t, len(idArray), 2)
+		assert.Equal(t, idArray[0], CadenceString(ids[0]))
+		assert.Equal(t, idArray[1], CadenceString(ids[2]))
```

Comment:
This is stored as a map and we are making assumptions about its order. AFAIK Cadence maps are un-ordered by definition. I think they used to happen to iterate in insert-order, but this is no longer true with some recent changes. 
```suggestion
		result := executeScriptAndCheck(t, b, templates.GenerateGetNonOperationalListScript(env), nil)

		idArray := result.(cadence.Array).Values
		assert.Equal(t, len(idArray), 2)
		assert.ElementsMatch(t, idArray, cadence.NewArray([]cadence.Value{CadenceString(ids[0]), CadenceString(ids[2])}))
```

Reply:
Good catch. Updated. 👍